packages/library-sync/src/commonMain/kotlin/io/realm/mongodb/internal/MutableSubscriptionSetImpl.kt

```diff
@@ -0,0 +1,85 @@
+package io.realm.mongodb.internal
+
+import io.realm.BaseRealm
+import io.realm.RealmObject
+import io.realm.internal.interop.RealmBaseSubscriptionSetPointer
+import io.realm.internal.interop.RealmInterop
+import io.realm.internal.interop.RealmMutableSubscriptionSetPointer
+import io.realm.internal.platform.realmObjectCompanionOrThrow
+import io.realm.internal.query.ObjectQuery
+import io.realm.mongodb.sync.MutableSubscriptionSet
+import io.realm.mongodb.sync.Subscription
+import io.realm.query.RealmQuery
+import kotlinx.atomicfu.AtomicRef
+import kotlinx.atomicfu.atomic
+import kotlin.reflect.KClass
+
+internal class MutableSubscriptionSetImpl<T : BaseRealm>(
+    realm: T,
+    nativePointer: RealmMutableSubscriptionSetPointer
+) : BaseSubscriptionSetImpl<T>(realm), MutableSubscriptionSet {
+
+    override val nativePointer: AtomicRef<RealmMutableSubscriptionSetPointer> = atomic(nativePointer)
```

Comment:
There is no reason for this to have an updatable native pointer!? It is only updated for `SubscriptionSetImpl` so the `BaseSubscriptionSetImpl` should probably just have a `val nativePointer: RealmBaseSubscriptionSetPointer` and then you could encapsulate the atomic updateable property in the getter of `SubscriptionSetImpl`